docs/interpreter_design_doc.md

```diff
@@ -0,0 +1,100 @@
+# Interpreter Design
+
+
+## Class Hierarchy
+
+Stablehlo programs are computations over Tensors (n-dimensional arrays), hence
+most of the input/output data values for an op are Tensors with the exception of
+constant
+[Attribute](https://docs.google.com/document/d/1GAAK_yHk7q2cZdJ3zBhsiNcrqzfpyjwQ-oWLD7q__Ss/edit#heading=h.5nem09i7p9va)
+values of primitive types (e.g., Int, float etc.).
```

Comment:
Are these attributes used as values, or they are part of value metadata?

Reply:
>> with the exception of constant Attributes
yes, the  tributes here are not used as values, but they are the metadata related to operand values.  I think this statement unnecessarily complicates the discussion. Will be revised.